frontend/src/api/index.ts

```diff
@@ -0,0 +1,17 @@
+import { RegistrationInput } from '../components/registrationForm'
+import axios from 'axios'
+
+// In terms of Design Patterns, you could call this either a Facade, a Gateway or an Adapter.
+
+export const api = {
+  posts: {
+    getPosts: () => {
+      return axios.get('http://localhost:3000/posts?sort=recent')
```

Comment:
using hardcoded URL usually is a bad idea as it makes it hard to deploy for example
I would at least move move domain and port to a separate constant
other approach would be to use Vite proxy mechanism to auto redirect all calls to `/api` path to backend
```
server: {
    proxy: {
      '/api': {
        target: 'http://localhost:3000',
        changeOrigin: true,
        rewrite: (path) => path.replace(/^\/api/, ''),
      },
    },
  },
```